### PR TITLE
[Layout] ortho_4x12: bredfield

### PR DESCRIPTION
**Pull request by bredfield**
## Description

Adds my 4x12 layout, used for both Planck and split style ortho boards.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: layouts/community/ortho_4x12/bredfield/keymap.c [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+// Keymap layers
+#define _BASE_LAYER 0
+#define _RAISE_LAYER 1
+#define _LOWER_LAYER 2
+#define _NUMPAD_LAYER 3
+#define _NAVIGATION_LAYER 4
+#define _UI_LAYER 5
+#define _KEYBOARD_LAYER 6
+
+// Key aliases
```
Comment (noroadsleft) [inline layouts/community/ortho_4x12/bredfield/keymap.c]: QMK defines aliases for these keycodes automatically.

https://github.com/qmk/qmk_firmware/blob/53a81fc2f681eda7267804eb990bc24c4ef58512/quantum/quantum_keycodes.h#L29-L31

If you use the predefined aliases in your keymap, you don't need to define either of these.

File: layouts/community/ortho_4x12/bredfield/keymap.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
+
+// Keymap layers
```
Comment (noroadsleft) [inline layouts/community/ortho_4x12/bredfield/keymap.c]: Recommended but optional: using an `enum` for your layer indexes.

This is programmatically equivalent to your current definitions:

```c
// Keymap layers
enum layer_names {
    _BASE_LAYER,
    _RAISE_LAYER,
    _LOWER_LAYER,
    _NUMPAD_LAYER,
    _NAVIGATION_LAYER,
    _UI_LAYER,
    _KEYBOARD_LAYER,
};
```

File: layouts/community/ortho_4x12/bredfield/keymap.c [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+#define GUI_RBR LGUI(KC_RBRC)
+#define GUI_ENT LGUI_T(KC_ENT)
+#define GUI_ESC LGUI_T(KC_ESC)
+#define GUI_TAB LGUI(KC_TAB) // application toggle
+#define GUI_GRV LGUI(KC_GRV) // window toggle
+#define GUI_SSF LGUI(LSFT(KC_3)) // Full screen shot
+#define GUI_SST LGUI(LSFT(KC_4)) // Targetted screen shot
+
+// Actions
+const uint16_t PROGMEM fn_actions[] = {
```
Comment (noroadsleft) [inline layouts/community/ortho_4x12/bredfield/keymap.c]: Lines 50-52 can be safely deleted. This function is deprecated and has been superseded by `process_record_user`.

**Comment by noroadsleft** [comment]
Additionally, we recommend against committing to your fork's `master` branch due to the potential for merge conflicts. It is highly recommended for QMK development - regardless of what is being done or where - to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your master updated here:

[**Best Practices: Your fork's master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_best_practices?id=your-fork39s-master-update-often-commit-never)

If you need any help with this just ask. You do not need to close this pull request, nor do you need to file a new one at this time.

File: layouts/community/ortho_4x12/bredfield/keymap.c (hunk 2 above)
Comment (drashna) [reply layouts/community/ortho_4x12/bredfield/keymap.c]: For this number of layers, I agree.  It will be a lot easier to manage, as an enum

**Comment by bredfield** [comment]
@noroadsleft @drashna Thanks for the review! I'll branch off of master for all future PRs.

File: layouts/community/ortho_4x12/bredfield/keymap.c [hunk 4]
```diff
@@ -155,10 +149,10 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
  */
 [_UI_LAYER] = LAYOUT_ortho_4x12(
-    GUI_TAB, ___X___, ___X___, ___X___, ___X___, ___X___, ___X___, ___X___, ___X___, GUI_MIN, GUI_EQL, ___X___,
-    GUI_GRV, GUI_1,   GUI_2,   GUI_3,   GUI_4,   GUI_5,   GUI_6,   GUI_7,   ___X___, GUI_LBR, GUI_RBR, ___X___,
-    _______, GUI_SSF, GUI_SST, ___X___, ___X___, ___X___, ___X___, ___X___, ___X___, ___X___, ___X___, ___X___,
-    _______, KC_MRWD, KC_MPLY, KC_MFFD, KC_BRID, _______, _______, KC_BRIU, KC_MUTE, KC_VOLD, KC_VOLU, _______
+    GUI_TAB, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, GUI_MIN, GUI_EQL, XXXXXXX,
+    GUI_GRV, GUI_1,   GUI_2,   GUI_3,   GUI_4,   GUI_5,   GUI_6,   GUI_7,   XXXXXXX, GUI_LBR, GUI_RBR, XXXXXXX,
+    _______, GUI_SSF, GUI_SST, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    _______, KC_MRWD, KC_MPLY, KC_MFFD, KC_VOLD, KC_MUTE, KC_MUTE, KC_VOLU, _______, _______, _______, _______
```
Comment (noroadsleft) [inline layouts/community/ortho_4x12/bredfield/keymap.c]: You removed your brightness control keys here, and I just wanted to make sure that was deliberate.

Everything else looks good.

Comment (bredfield) [reply layouts/community/ortho_4x12/bredfield/keymap.c]: Yep, wasn't working with Mac OS. Thanks for checking

Comment (drashna) [reply layouts/community/ortho_4x12/bredfield/keymap.c]: IIRC, that's because the hardware ID ***must*** match to a known apple keyboard.

**Pull request merged**